### PR TITLE
2031: Refactor getDate method

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description

The getDate method has duplicate code in web and native. I moved the shared logic of choosing the right icon to the api-client. Only selecting the platform specific icons stays in web and native.

### CODE REVIEW

File: api-client/src/models/DateModel.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { RRule as RRuleType } from 'rrule'
 
 const MAX_RECURRENCE_YEARS = 5
 
+export type EventIcon = 'CalendarTodayRecurringIcon' | 'CalendarRecurringIcon' | 'CalendarTodayIcon'
```
Comment (steffenkleinle) [inline api-client/src/models/DateModel.ts:6]:
```suggestion
export type DateIcon = 'CalendarTodayRecurringIcon' | 'CalendarRecurringIcon' | 'CalendarTodayIcon'

```

File: api-client/src/models/DateModel.ts [hunk 2]
```diff
@@ -101,6 +102,22 @@ class DateModel {
     return `${localizedStartDate} - ${localizedEndDate.toFormat(format)}`
   }
 
+  getDateIcon(): { icon: EventIcon; label: string } | null {
```
Comment (steffenkleinle) [inline api-client/src/models/DateModel.ts:105]:
```suggestion
  getDateIcon(): { icon: DateIcon; label: string } | null {
```

File: native/src/components/EventListItem.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { SvgProps } from 'react-native-svg'
 import styled from 'styled-components/native'
 
 import { DateModel, EventModel, parseHTML } from 'api-client'
+import { EventIcon } from 'api-client/src/models/DateModel'
```
Comment (steffenkleinle) [inline native/src/components/EventListItem.tsx:7]: Please export in index.ts and use import from 'api-client'

File: native/src/components/EventListItem.tsx [hunk 4]
```diff
@@ -25,19 +26,18 @@ const Description = styled.Text`
...
-
-  if (isRecurring && isToday) {
-    return { icon: CalendarTodayRecurringIcon, label: 'todayRecurring' }
-  }
-  if (isRecurring) {
-    return { icon: CalendarRecurringIcon, label: 'recurring' }
-  }
-  if (isToday) {
-    return { icon: CalendarTodayIcon, label: 'today' }
+  const nativeIcons: { [key in EventIcon]: JSXElementConstructor<SvgProps> } = {
```
Comment (steffenkleinle) [inline native/src/components/EventListItem.tsx:29]: I think the prefix `native` is confusing bc there are no web icons here, just `icons` would be enough imo:
```suggestion
  const icons: { [key in DateIcon]: JSXElementConstructor<SvgProps> } = {
```

File: web/src/components/EventListItem.tsx [hunk 5]
```diff
@@ -38,19 +39,18 @@ const getEventPlaceholder = (path: string): string => {
...
-
-  if (isRecurring && isToday) {
-    return { icon: CalendarTodayRecurringIcon, tooltip: 'todayRecurring' }
-  }
-  if (isRecurring) {
-    return { icon: CalendarRecurringIcon, tooltip: 'recurring' }
-  }
-  if (isToday) {
-    return { icon: CalendarTodayIcon, tooltip: 'today' }
+  const webIcons: { [key in EventIcon]: string } = {
```
Comment (steffenkleinle) [inline web/src/components/EventListItem.tsx:42]: Same as for native:
```suggestion
  const icons: { [key in DateIcon]: string } = {
```

**Review by steffenkleinle — APPROVED** [review]
Nice refactoring, thanks :)

**Review by LeandraH — COMMENTED** [review]
I'm getting the following error when running `yarn start` in /web:
WARNING in ../../api-client/src/index.ts 61:0-68
export 'DateIcon' (reexported as 'DateIcon') was not found in './models/DateModel' (possible exports: default)

**Comment by ztefanie** [comment]
> I'm getting the following error when running `yarn start` in /web: WARNING in ../../api-client/src/index.ts 61:0-68 export 'DateIcon' (reexported as 'DateIcon') was not found in './models/DateModel' (possible exports: default)

Thanks for testing again, fixed it.
I think a lot of our errors happen, when we just change small things after other team members reviewed and tested. But, this error was catched by e2e tests.

**Review by LeandraH — APPROVED** [review]

**Comment by LeandraH** [comment]
Nice! Tested in emulated iOS and Firefox

**Pull request merged**